src/sorted_vector.jl

```diff
@@ -0,0 +1,52 @@
+"""
+A `SortedVector` behaves like a standard Julia `Vector`, except that its elements are stored in sorted order, with an optional function `by` that determines the sorting order in the same way as `Base.searchsorted`.
+"""
+immutable SortedVector{T, F<:Function}
+    data::Vector{T}
+    by::F
+
+    function SortedVector(data::Vector{T}, by::F)
+        new(sort(data), by)
+    end
+end
+
+
+SortedVector{T,F}(data::Vector{T}, by::F) = SortedVector{T,F}(data, by)
+SortedVector{T}(data::Vector{T}) = SortedVector{T,typeof(identity)}(data, identity)
+
+function show(io::IO, v::SortedVector)
+    print(io, "SortedVector($(v.data))")
+end
+
+
+
+getindex(v::SortedVector, i::Int) = v.data[i]
+length(v::SortedVector) = length(v.data)
+
+
+function insert!{T}(v::SortedVector{T}, i::Int, x::T)
```

Comment:
Shouldn't use `::T`, just leave `push!` do the conversion automatically if possible. Same below.

Though this method should probably throw an error, since it doesn't respect the contract of inserting `x` at position `i`.